### PR TITLE
SCALE codec tests

### PR DESCRIPTION
**Pull request by nadeemb53**
Signed-off-by: nadeemb53 <nadeemb53@gmail.com>

### CODE REVIEW

File: contracts/ethereum/src/core/node-codec/ScaleCodec.sol [hunk 1]
```diff
@@ -1,9 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.17;
 
-import "../../interfaces/ICodec.sol";
-
-contract ScaleCodec is ICodec {
+contract ScaleCodec {
```
Comment (blasrodri) [inline contracts/ethereum/src/core/node-codec/ScaleCodec.sol:4]: why not referencing the interface any more?

**Comment by blasrodri** [comment]
Can you try adding some more complex structures (perhaps build them using Rust) and bring them as hex here and ensure that the encoding works also with these?

File: contracts/ethereum/src/core/node-codec/ScaleCodec.sol (hunk 1 above)
Comment (nadeemb53) [reply contracts/ethereum/src/core/node-codec/ScaleCodec.sol:4]: not needed anymore. It's in use by NodeCodec contract though.

**Comment by nadeemb53** [comment]
> Can you try adding some more complex structures (perhaps build them using Rust) and bring them as hex here and ensure that the encoding works also with these?

as we're dealing with u32 at max, the decoded value would not exceed 2^32-1 (4294967295). But I'll try to add more diverse test cases.

**Pull request merged**